packages/create/src/validation/validation.ts

```diff
@@ -0,0 +1,33 @@
+import { existsSync } from 'fs';
+import type { Editor } from 'mem-fs-editor';
+import { join } from 'path';
+import type { EditorWithDump } from '../types';
+
+/**
+ * Validate base path of app, throw error if file is missing.
+ *
+ * @param basePath - base path of the app, where package.json and ui5.yaml resides
+ */
+export function validateBasePath(basePath: string): void {
+    const packageJsonPath = join(basePath, 'package.json');
+    if (!existsSync(packageJsonPath)) {
+        throw Error(`Required file '${packageJsonPath}' does not exist.`);
+    }
+    const ui5YamlPath = join(basePath, 'ui5.yaml');
+    const webappPath = join(basePath, 'webapp');
+    if (!existsSync(ui5YamlPath) && !existsSync(webappPath)) {
```

Comment:
The `ui5.yaml` is required based on this validation function, however, you have code for creating the `ui5-mock.yaml` that handles the case of no `ui5.yaml`.
So is it required or not?

Reply:
No, there is an `&&`. File `ui5.yaml` is only required if there is no `webapp` path.

Reply:
Thanks for explaining and and or conditions to me :/ . I missed that one.